src/mpid/ch4/shm/xpmem/xpmem_pre.h

```diff
@@ -26,6 +40,15 @@ typedef struct MPIDI_XPMEM_seg {
     MPIR_cc_t refcount;         /* reference count of this seg */
 } MPIDI_XPMEM_seg_t;
 
+typedef struct MPIDI_XPMEM_cnt {
+    MPIR_OBJECT_HEADER;
+    OPA_int_t counter;
+    uint32_t free_flag;         /* when receiver copies all chunks, sender should
+                                 * set it as 1 to notify receiver that sender has
+                                 * completed cooperative copy and counter can be
+                                 * released. */
+} MPIDI_XPMEM_cnt_t;
```

Comment:
We should use `bool` type for the `free_flag`. And the value will be `true|false` instead of `1|0`.